tst/hashmap.tst

```diff
@@ -36,6 +36,35 @@ true
 gap> Set(List(KeyValueIterator(hashmap))) = List([1..1000],i->[i,i^2]);
 true
 
+# Check initial construction
+gap> Set(List(KeyValueIterator(HashMap([])))) = [];
```

Comment:
```suggestion
gap> Set(KeyValueIterator(HashMap([]))) = [];
```
Set(List(...)) should be redundant. Well, *should* be -- but if it isn't then IMHO that's a bug somewhere?

Reply:
OK turns out that only `List` works for iterators, but not `Set` or `AsList`. Actually as far as I could tell, even `List` is not documented for iterators. I'll file an issue